src/Microsoft.AspNetCore/WebHost.cs

```diff
@@ -182,6 +182,16 @@ public static IWebHostBuilder CreateDefaultBuilder(string[] args)
                     logging.AddConsole();
                     logging.AddDebug();
                 })
+                .ConfigureServices((hostingContext, services) =>
+                {
+                    var hosts = hostingContext.Configuration["AllowedHosts"]?.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
+                    services.AddHostFiltering(options =>
+                    {
+                        options.AllowedHosts = (hosts?.Length > 0 ? hosts : new[] { "*" });
+                    });
+
+                    services.AddTransient<IStartupFilter, HostFilteringStartupFilter>();
```

Comment:
Can you make sure this middleware is optimized for nooping?

Reply:
It is. https://github.com/aspnet/BasicMiddleware/blob/2f92fc1d1eaa4033e5b02581da0018dcf8511ea1/src/Microsoft.AspNetCore.HostFiltering/HostFilteringMiddleware.cs#L150